src/mol-gl/shader/shadows.frag.ts

```diff
@@ -80,20 +80,19 @@ float screenSpaceShadow(const in vec3 position, const in vec3 lightDirection, co
         rayCoords = uProjection * vec4(rayPos, 1.0);
         rayCoords.xyz = (rayCoords.xyz / rayCoords.w) * 0.5 + 0.5;
 
-        if (outsideBounds(rayCoords.xy))
-            return 1.0;
+        //if (outsideBounds(rayCoords.xy))
+        //    return 1.0;
 
         // Compute the difference between the ray's and the camera's depth
         float depth = getDepth(rayCoords.xy);
-        float viewZ = getViewZ(depth);
-        float zDelta = rayPos.z - viewZ;
+        float sampleViewZ = screenSpaceToViewSpace(vec3(rayCoords.xy, depth), uInvProjection).z;
+        // float viewZ = getViewZ(depth);
+        float zDelta = rayPos.z - sampleViewZ;
 
         if (zDelta < uTolerance) {
             occlusion = 1.0;
-
             // Fade out as we approach the edges of the screen
-            occlusion *= screenFade(rayCoords.xy);
-
+            // occlusion *= screenFade(rayCoords.xy);
```

Comment:
I would rather keep some fading because it avoids artifacts like the one below.

![image](https://github.com/molstar/molstar/assets/272250/19f18bb8-151a-4521-a248-27375c4ec601)


An improvement could be to take the current lights direction into account for fading.

Reply:
I re-enable it. I will keep in mind your suggestion for the fading direction.

Reply:
still there, did you push everything?

Reply:
not sure what happen, latest push has everything.